lib/lita/handlers/jira_issues.rb

```diff
@@ -34,53 +36,62 @@ def handle_key(response, key)
       def issue_details(data)
         key = data[:key]
         data = data[:fields]
-        issue = summary(key, data)
-        issue << status(data)
-        issue << assignee(data)
-        issue << reporter(data)
-        issue << fix_version(data)
-        issue << priority(data)
-        issue << issue_link(key)
-      end
-
-      def summary(key, data)
-        "[#{key}] #{data[:summary]}"
+        
+        # build out the response from the configured format
+        text = String.new(config.format)
+        text.sub!('%I', key.upcase)
```

Comment:
I like the terseness of just `%<letter>`, but I also like Ruby's idiomatic [string format using a hash](http://ruby-doc.org/core-2.2.0/String.html#method-i-25). So in this case it would look something like:

```ruby
config :format, default: "[%{I}] %{t}\nStatus: %{s}, assigned to %{a}, rep. by %{r}, fixVersion: %{v}, priority: %{P}\n%{U}", type: String
...
text = config.format % {
  I: key.upcase,
  i: key.downcase,
  S: status.upcase
...
```

You could even take it further and make it the formatter a full name like:

```ruby
FORMAT = <<-FORMATTER
[%{KEY}] %{summary}
Status: %{status}, assigned to %{assignee}, rep. by %{reporter}, fixVersion: %{fix_version}, priority: %{priority}
%{issue_link}"
  FORMATTER

config :format, default: FORMAT, type: String
```

Where an app upper case key makes the data all upper case.